buffer/src/main/java/io/netty5/buffer/api/unsafe/UnsafeBuffer.java

```diff
@@ -1280,14 +1280,21 @@ private RuntimeException readAccessCheckException(int index, int size) {
         return outOfBounds(index, size);
     }
 
-    private RuntimeException writeAccessCheckException(int index, int size) {
+    private void writeAccessCheckException(int index, int size, boolean mayExpand) {
```

Comment:
nit: consider renaming to something like `checkWriteAccess(...)` as it is throwing now directly